psh/runfile/runfile.c

```diff
@@ -80,7 +80,20 @@ int psh_runfile(int argc, char **argv)
 	/* Take back terminal control */
 	tcsetpgrp(STDIN_FILENO, getpgid(getpid()));
 
-	return ((pid > 0) && (ret >= 0)) ? WEXITSTATUS(status) : EXIT_FAILURE;
+	/* Determine the status of fhe file */
+	if (!((pid > 0) && (ret >= 0))) {
+		return_value = EXIT_FAILURE;
+	}
+	if (0 != WIFSIGNALED(status)) {
```

Comment:
should be `else if`, otherwise `status == 0` and we will jump to final `else`.

Please also note that the comment above is invalid (we're not checking the status of the file, we're checking if vfork / waitpid finished successfully)